src/multio/action/average-rate/AverageRate.cc

```diff
@@ -0,0 +1,89 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+#include "AverageRate.h"
+
+#include "eckit/config/Configuration.h"
+
+#include "multio/LibMultio.h"
+#include "multio/message/Glossary.h"
+
+namespace multio::action::average_rate {
+
+namespace {
+Param2ParamMap getMappings() {
```

Comment:
Maybe split in `makeMappings` and `getMappings`, where `getMappings` performs:

```
const Param2ParamMap& getMappings() {
     static const Param2ParamMap map = getMappings();
     return map;
}
```

As this is important logic, I don't think it should be configurable. I'd even go that far to hardcode it... that's what we do in mars2mars and mars2grib now.

Just the mapping can be tested in isolation without testing the action.